### ISSUE TITLE
Consider merging hcloud-upload-image in this tool

### ISSUE THREAD

**Issue by travier** [hetznercloud/cli#825]
### TL;DR

Hey folks,

For Fedora CoreOS, we are looking at using https://github.com/apricote/hcloud-upload-image to document how to setup Fedora CoreOS on Hetzner Cloud.

https://github.com/apricote/hcloud-upload-image is currently marked as not officially supported by Hetzner / not an official Hetzner Cloud product.

We would appreciate if you could consider making it an official tool or merging it in this tool.

That would increase our (and our user's) confidence in this process and thus let us display Hetzner as a supported platform for Fedora CoreOS.

Thanks!

### Expected behavior

https://github.com/apricote/hcloud-upload-image becomes an official tool in some form.

**Comment by PhrozenByte** [comment]
I'd like to add that as an alternative, the possibility to directly upload snapshot images via the API would be much appreciated. [`apricote/hcloud-upload-image`](https://github.com/apricote/hcloud-upload-image) is a workaround after all.

**Comment by apricote** [comment]
Technically this is not an issue, I did a [very quick port](https://github.com/hetznercloud/cli/compare/main...upload-image) last week after the point was made in your docs PR.

I am not sure yet if we want to officially support this, as the risk with leftover paid servers is there. Also if an official API is added we would immediately deprecate/remove the workaround.

I have asked the team responsible for Servers and Images for their opinion on this and also added votes to the "Upload Image API" card in our internal feedback tracker.

**Comment by dustymabe** [comment]
> I have asked the team responsible for Servers and Images for their opinion

I'm wondering if they had any opinion that could be shared.

**Comment by apricote** [comment]
Bad news, we do not want to provide official support for the workaround CLI for the foreseeable future.

The teams opinion matched my own:

> I am not sure yet if we want to officially support this, as the risk with leftover paid servers is there. Also if an official API is added we would immediately deprecate/remove the workaround.

If we add an official API, it will be immediately supported in the CLI & hcloud-go.
